service/proxy/src/main/java/org/kaazing/gateway/service/proxy/ProxyServiceHandler.java

```diff
@@ -106,6 +100,12 @@ public void initializeSession(IoSession connectSession, ConnectFuture future) {
             }
 
             super.sessionOpened(acceptSession);
```

Comment:
No.  If the extension needs to put something on the filter chain then the code needs to be added first so that extensions can just add themselves to the end.  Basically this is saying that the Gateway will do its work first, and then extensions will get a shot.